design/one-pager-consuming-k8s-clusters.md

```diff
@@ -62,21 +64,32 @@ referred to as a "Bring Your Own Cluster" scenario.
 ## Proposal
 
 This proposal can be separated into two main categories:
-1. The implementation of a new `WorkloadTarget` custom resource.
+1. The implementation of a new `KubernetesTarget` custom resource and
+   corresponding controller.
 1. The modification of the scheduling behavior for `KubernetesApplication`.
 
-### WorkloadTarget
+### KubernetesTarget
 
-The purpose of the `WorkloadTarget` resource is to effectively "publish"
+The purpose of the `KubernetesTarget` resource is to effectively "publish"
 Kubernetes clusters for usage in a namespace. It is namespace-scoped and can
 only reference Kubernetes cluster managed resources or the Kubernetes `Provider`
-type, both of which are cluster-scoped. An example of a `WorkloadTarget` in
-namespace `my-app-team` that is referencing a GKE cluster managed resource could
-look as follows:
+type, both of which are cluster-scoped. The creation of a `KubernetesTarget`
+resource in a namespace will be watched by a single new secret propagation
+controller, which will get the `Secret` referenced by either the Kubernetes
+cluster managed resource or Kubernetes `Provider` and propagate it to the
```

Comment:
Is there a way to make this reference exclusive to `Kubernetes Provider` type? Like if a user who is able to create a `KubernetesTarget` shouldn't be able to pull in credentials of a `AWS Provider` or other cloud provider secrets.